src/movies.js

```diff
@@ -1,25 +1,101 @@
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(array) {
+  const allDirectors = array.map(function (arr) {
+      return arr.director
+    });
+  return allDirectors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(array) {
+  let howManyDramaMovies = 0;
+  const allSpielbergDrama = array.filter(function (arr) {
+    return (arr.director === 'Steven Spielberg' && arr.genre.indexOf('Drama') !==-1)
+  });
+  howManyDramaMovies = allSpielbergDrama.length
+  return howManyDramaMovies
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(array) {
+  if (array.length === 0) return 0;
+  const sumScore = array.reduce(function (acc, val) {
+    if (val.score !== undefined){
+    return acc += val.score}
+    else return acc
+  }, 0)
+  let sumAvgScore = 0;
+  sumAvgScore = sumScore/array.length;
+  return parseFloat(sumAvgScore.toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+
+function dramaMoviesScore(array) {
+  let allDramaMovies = array.filter(function (arr){
+    return arr.genre.includes('Drama')} );    
+  if (allDramaMovies.length === 0) return 0;
+  let allDramaMoviesScore = allDramaMovies.reduce(function (acc, value){
+    return acc+value.score;
+  }, 0)
+  let avgScore = allDramaMoviesScore/allDramaMovies.length
+  return parseFloat(avgScore.toFixed(2))
+} 
+  
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(array) {
+  let newArray = array.slice()
+  let orderedByYear= newArray.sort(function(a, b) {
+    if (a.year < b.year) {
+      return -1;
+    }
+    if (a.year> b.year) {
+      return 1;
+    }  
+  });  
+  orderedByYear = orderedByYear.sort(function(a,b){
+    if (a.year === b.year){
+     return a.title.localeCompare(b.title);
+    }
+  })
+ 
+  return orderedByYear;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+function orderAlphabetically(array) {
+  
+  let allMovieTitles = array.map(function(arr){
+    return arr.title;
+  }); 
+  let sortAlphabetically= allMovieTitles.sort(function (a,b){
+    return a.localeCompare(b);
+  })
+  return sortAlphabetically.slice(0,20)
+}
+
 
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes() {}
+function turnHoursToMinutes(arr) {
+  let copyHoursToMinutes =JSON.parse(JSON.stringify(arr));
+  
+  let minutes = 0;
+  let hours = 0;
+  
+  copyHoursToMinutes.map(function (array) {
+  
+  hours = parseInt(array.duration.split(' ') || 0);
```

Comment:
The `split` function splits a string and returns the result into an array. `parseInt` expects a string. I think the reason the test case passes is that the array here is converted to a string by JavaScript (concatenating all its elements on one string) and `parseInt` picks the number out of this string. 

Another point here is that map always runs on an array and returns a new array. Here you are missing to return an array.

I think a better way is to use slice to pick the number before the 'h' for the minutes and after the space for the minutes. Check the solution Jan posted on Github.